src/main/java/com/iexec/resultproxy/authorization/AuthorizationService.java

```diff
@@ -109,4 +114,34 @@ public String getChallengeForWorker(WorkerpoolAuthorization workerpoolAuthorizat
                 workerpoolAuthorization.getChainTaskId(),
                 workerpoolAuthorization.getEnclaveChallenge());
     }
+
+    // region workerpool authorization cache
+    public boolean checkEnclaveSignature(ResultModel model, String walletAddress) {
+        final String chainTaskId = model.getChainTaskId();
+        final String wpAuthKey = String.join("-", chainTaskId, walletAddress);
+        String resultHash = HashUtils.concatenateAndHash(chainTaskId, model.getDeterministHash());
```

Comment:
several variables should be able to pass as final in this method

Reply:
done